modules/isc_dhcpd/lease.go

```diff
@@ -0,0 +1,137 @@
+package isc_dhcpd
+
+import (
+	"errors"
+	"net"
+	"os"
+	"time"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+	leases "github.com/npotts/go-dhcpd-leases"
+)
```

Comment:
i dont think we need `github.com/npotts/go-dhcpd-leases`

Please make `parse` a separate file, it contains

```go
package isc_dhcpd

type Lease struct {
	IP           string
	Ends         string
	BindingState string
}

func parseDHCPDLeases(leases []Lease, filepath string) ([]Lease, error) {
	...
}

```

It should have **only this logic, no any additional**. And add `parse_test.go` file and put here tests for different inputs (files/cases:ipv4 leases, ipv6 leases, backup leases (has no ends), static leases - end is `never`, etc)

---

`parseDHCPDLeases` takes `[]Lease` and returns `[]Lease`, yes, this is needed to reuse same slice.

`Ends` is string because sometimes it is `never`.

Reply:
I changed the parser, now we have our parser doing what we need.